### PR TITLE
Add `align-self: center` to buttons for improved rendering in flex containers

### LINKED ISSUE #34833: button size not respected when direct child of div with d-flex
When placing a button with size classes as direct child of a div with d-flex, size is not respected

    <div class="d-flex flex-row">
      <button type="button" class="m-2 btn btn-primary">Default</button>
      <button type="button" class="m-2 btn btn-primary btn-sm">Small</button>
      <button type="button" class="m-2 btn btn-primary btn-md">Medium</button>
      <button type="button" class="m-2 btn btn-primary btn-lg">Large</button>
    </div>

produces
image

While

    <div class="d-flex flex-row">
      <div>
        <button type="button" class="m-2 btn btn-primary">Default</button>
        <button type="button" class="m-2 btn btn-primary btn-sm">Small</button>
        <button type="button" class="m-2 btn btn-primary btn-md">Medium</button>
        <button type="button" class="m-2 btn btn-primary btn-lg">Large</button>
      </div>
    </div>

produces
image

Not sure if it's a bug, or more likely that I don't understand it properly.

- Bootstrap 5.1.0
- Operating system and version (Windows, macOS, Android, iOS)
       Windows 10 21H1
- Browser and version (Chrome, Firefox, Safari, Microsoft Edge, Opera, Android Browser)
    Microsoft Edge : Version 92.0.902.78
    Chrome : Version 92.0.4515.159 
    Firefox : 91.0.2 (64-bit)

### PR DESCRIPTION
**Pull request by zaidjawed**
Fixes #34833

When placing a button with size classes as direct child of a div with d-flex, size is not respected

```
<div class="d-flex flex-row">
  <button type="button" class="m-2 btn btn-primary">Default</button>
  <button type="button" class="m-2 btn btn-primary btn-sm">Small</button>
  <button type="button" class="m-2 btn btn-primary btn-md">Medium</button>
  <button type="button" class="m-2 btn btn-primary btn-lg">Large</button>
</div>
```
Produces
131220125-75cef342-d108-442d-811b-25a2ecb679a1

While

```
<div class="d-flex flex-row">
  <div>
    <button type="button" class="m-2 btn btn-primary">Default</button>
    <button type="button" class="m-2 btn btn-primary btn-sm">Small</button>
    <button type="button" class="m-2 btn btn-primary btn-md">Medium</button>
    <button type="button" class="m-2 btn btn-primary btn-lg">Large</button>
  </div>
</div>
```

Produces
131220163-0b761e9f-f518-4d68-bc99-43052cc18992

Solution:

According to [W3Schools](https://www.w3schools.com/cssref/css3_pr_align-items.asp) 
the default value of align-items is **stretch** which might be causing this issue.

I simply added **align-self to center** to give a default value to center if the buttons are a direct child of flex.

### CODE REVIEW

File: scss/_buttons.scss [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 .btn {
   display: inline-block;
+  
+  //button size not respected when direct child of div with d-flex #34833
+  align-self: center;
+
```
Comment (ffoodd) [inline scss/_buttons.scss:10]: Please drop the line breaks and comments, this is not needed.

**Review by ffoodd — CHANGES_REQUESTED** [review]
Feels safe to me, thanks for submitting this :)

**Review by ffoodd — APPROVED** [review]

**Review by mdo — APPROVED** [review]
I'm into it!

**Comment by wKovacs64** [comment]
This was a breaking change. Can't you just apply `align-items-center` on your flex container instead?

```
<div class="d-flex flex-row align-items-center">
  <button type="button" class="m-2 btn btn-primary">Default</button>
  <button type="button" class="m-2 btn btn-primary btn-sm">Small</button>
  <button type="button" class="m-2 btn btn-primary btn-md">Medium</button>
  <button type="button" class="m-2 btn btn-primary btn-lg">Large</button>
</div>
```

**Comment by andreas-venturini** [comment]
This broke most of the button styles in our application. I strongly suggest highlighting this change more prominently in the Changelog (possibly even mark as a breaking change). This is not what we would've expected of a minor patch release.

Anyway, keep up the good work :v:

**Comment by andreas-venturini** [comment]
@ffoodd done https://github.com/twbs/bootstrap/issues/35126

Thinking about this some more, I think you should consider reverting (I'm not disagreeing with this change per se, it might just be a bit too much for a minor patch release)

**Comment by amshkv** [comment]
This change was unexpected and broke the part of the interface where the parent was `d-flex` with `align-items-start`
Now we need to redefine all `btn` in this layout.
Especially unexpected for `flex-column`
image

**Pull request merged**